src/context/auth-context.tsx

```diff
@@ -0,0 +1,186 @@
+import React, {createContext, ReactNode, useContext, useEffect, useRef, useState} from "react";
+import {invoke} from '@tauri-apps/api/core';
+import {getCurrentWindow, type WindowOptions} from '@tauri-apps/api/window';
+import {WebviewWindow} from "@tauri-apps/api/webviewWindow";
+import {settings} from '../tauri-store/settings.ts'; // Adjust the import path as necessary
+import {Event} from "@tauri-apps/api/event";
+import {cancel} from "@fabianlars/tauri-plugin-oauth";
+import {AuthenticationResponse} from "../model/authentication-response.ts";
+import {SecretVariables} from "../model/secret-variables.ts";
+
+export interface AuthContextType {
+    authResponse: AuthenticationResponse | null;
+    loggedOut: boolean;
+    isLoggingIn: boolean;
+    fetchSecretsError: string | null;
+    login: () => Promise<void>;
+    logout: () => Promise<void>;
+}
+
+const AuthContext = createContext<AuthContextType | null>(null);
```

Comment:
Tror denne skulle hatt en default-verdi av AuthContextType slik at man slipper å caste til rett type når man bruker useAuth(). Kanskje man kan skrive det slik:
```suggestion
const AuthContext = createContext<AuthContextType>({} as AuthContextType);
```
Har ikke testet det men tror det burde være mulig.